html/semantics/forms/form-submission-0/implicit-submission.optional.html

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<link rel="help" href="https://html.spec.whatwg.org/C/#implicit-submission">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="./resources/targetted-form.js"></script>
+<body>
+<script>
+// This test file is "optional" because triggering implicit submission by
+// "Enter" key is not standardized.
+
+const ENTER = '\uE007';
+
+promise_test(async () => {
+  let form = populateForm('<input name=text value=abc><input name=submiButton type=submit>');
+  let event;
+  form.text.focus();
+  form.addEventListener('submit', e => { event = e; });
+  await test_driver.send_keys(form.text, ENTER);
+  assert_true(event.bubbles);
+  assert_true(event.cancelable);
+  assert_equals(event.submitter, form.submitButton);
+  assert_true(event instanceof SubmitEvent);
+}, 'Submit event with a submit button');
+
+promise_test(async () => {
+  let form = populateForm('<input name=text value=abc>');
+  let event;
+  form.text.focus();
+  form.addEventListener('submit', e => { event = e; });
+  await test_driver.send_keys(form.text, ENTER);
+  assert_true(event.bubbles);
+  assert_true(event.cancelable);
+  assert_equals(event.submitter, form);
```

Comment:
I guess this is fine, but I wonder if this will lead to weirdness with `new FormData()`. I guess we'd just ignore the element?

Reply:
I'm ok to change the `submitter` attribute so that it will be `null` if the _submitter_ is a &lt;form>.

[`requestSubmit(submitter)`](https://html.spec.whatwg.org/multipage/forms.html#dom-form-requestsubmit) doesn't accept &lt;form> as a _submitter_.  Dropping &lt;form> from web-exposed submitter might be easier to understand.

What do you think?

Reply:
Yeah, that's what I was thinking as well.

Reply:
Hmm. I like the simplicity of exposing the form, but I also see the argument for consistency with requestSubmit(). I think we should re-read the thread to make sure that null submitter when the submitter is the form still meets all the use cases that came up.